cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -586,6 +586,15 @@ func (dc *DownstreamController) deleteFromConfigMap(device *v1alpha1.Device) {
 		dc.deleteFromDeviceProfile(device, nodeConfigMap)
 		// store new config map
 		dc.configMapManager.ConfigMap.Store(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], nodeConfigMap)
+
+		// no device is bound to the configMap, then remove the configMap directly.
```

Comment:
Can we move line line 588 below this if to reduce 1 extra store operation before delete ?

Reply:
sure, will do later.